darwin/dataset/remote_dataset_v2.py

```diff
@@ -552,3 +558,64 @@ def _build_image_annotation(
         self, annotation_file: AnnotationFile, team_name: str
     ) -> Dict[str, Any]:
         return build_image_annotation(annotation_file, team_name)
+
+    def register(
+        self,
+        object_store: ObjectStore,
+        storage_keys: List[str],
+        fps: Optional[str] = None,
+        multi_planar_view: bool = False,
+        preserve_folders: bool = False,
+    ) -> Dict[str, List[str]]:
+        items = []
+        for storage_key in storage_keys:
+            file_type = get_external_file_type(storage_key)
+            if not file_type:
+                raise TypeError(
+                    f"Unsupported file type for the following storage key: {storage_key}.\nPlease make sure your storage key ends with one of the supported extensions:\n{SUPPORTED_EXTENSIONS}"
+                )
+            item = {
+                "path": parse_external_file_path(storage_key, preserve_folders),
+                "type": file_type,
+                "storage_key": storage_key,
+                "name": (
+                    storage_key.split("/")[-1] if "/" in storage_key else storage_key
+                ),
+            }
+            if fps and file_type == "video":
+                item["fps"] = fps
+            if multi_planar_view and file_type == "dicom":
+                item["extract_views"] = "true"
+            items.append(item)
+
+        # Do not register more than 500 items in a single request
+        chunked_items = [items[i : i + 500] for i in range(0, len(items), 500)]
+        print(f"Registering {len(items)} items in {len(chunked_items)} batch(es)")
```

Comment:
we could use a generator expression instead of a list comprehension for `chunked_items`. This will create the chunks on-the-fly during the iteration, which can save memory if items is very large:

```suggestion
        # Do not register more than 500 items in a single request
        chunk_size = 500
        chunked_items = (items[i : i + chunk_size] for i in range(0, len(items), chunk_size))
        print(f"Registering {len(items)} items in chunks of {chunk_size} items...")
```

_Python also introduced a [itertools.batched](https://docs.python.org/3.12/library/itertools.html#itertools.batched) function in python3.12 (they also added an 'equivalent function' in the same docs as well for older python versions - maybe we can use that? - just an optimisation to our chunking approach)_

Reply:
Great suggestion! Sadly at the moment we're support `[>=3.8.0, <3.12](https://pypi.org/project/darwin-py/)`, I think moving to 3.12 for this change would be too disruptive. What do you say to sticking with the generator expression?